x-pack/plugins/infra/public/lib/settings_locale_storage.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+function parseJsonOrDefault<Obj>(value: string | null, defaultValue: Obj): Obj {
```

Comment:
nit (more common to see generic type names as T or TSomething):
```suggestion
function parseJsonOrDefault<TObj>(value: string | null, defaultValue: TObj): TObj {
```